src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/checkout/CheckoutPaymentsResult.java

```diff
@@ -112,16 +117,23 @@ public PurchaseResult toPurchaseResult() {
             additionalData.put(AdyenPaymentPluginApi.PROPERTY_MERCHANT_ACCOUNT_CODE, merchantAccount);
         }
 
-        if(formParameter != null) {
+        if(paymentInfo.getIdentifierMap() != null && !paymentInfo.getIdentifierMap().isEmpty()) {
```

Comment:
can we use CollectionUtils.isNotEmpty()?

Reply:
That doesn't work with Map. Map is not a collection, it has different hierarchy.